osdk/src/commands/build/mod.rs

```diff
@@ -231,6 +231,36 @@ fn build_kernel_elf(
 
     let mut command = cargo();
     command.env_remove("RUSTUP_TOOLCHAIN");
+    command.env(
+        "ASTER_COMPILE_BY",
```

Comment:
OSTD and OSDK are positioned as a crate and tool for Rust OS development in general, not specific to Asterinas, although they origin from Asterinas. So these environment variable names should not be prefixed with "ASTER_". I suggest renaming them to:
* `ASTER_COMPILE_BY` --> `OSDK_BUILD_USERNAME`
* `ASTER_COMPILE_HOST ` --> `OSDK_BUILD_HOSTNAME`
* `ASTER_COMPILER ` --> `OSDK_BUILD_RUSTC`
* `ASTER_COMPILE_VERSION ` --> `OSDK_BUILD_TIMESTAMP`